modules/seqexec/server/src/main/scala/seqexec/server/gmos/Gmos.scala

```diff
@@ -59,6 +60,31 @@ abstract class Gmos[F[_]: Sync, T<:GmosController.SiteDependentTypes](controller
     }
   }
 
+  // The OT lets beams up to G but in practice it is always A/B
+  private val BeamLabels = List('A, 'B, 'C, 'D, 'E, 'F, 'G)
+
+  private def nsPosition(config: Config, sc: Int): Either[ExtractFailure, Vector[NSPosition]] = {
+    (for {
+      i <- 0 to scala.math.min(BeamLabels.length, sc) - 1
+    } yield {
+      for {
+        s <- BeamLabels.lift(i).toRight(ContentError(s"Unknwon label at position $i"))
+        p <- config.extractAs[String](INSTRUMENT_KEY / s"nsBeam${s.name}-p").map(_.toDouble)
+        q <- config.extractAs[String](INSTRUMENT_KEY / s"nsBeam${s.name}-q").map(_.toDouble)
```

Comment:
The `seqexec.model.TelescopeOffset` predates this PR, but we've been working hard to keep floating point values out of the model. `gsp-math` has an `Offset` class that stores the offset value unambiguously and perhaps in a future task it could be used instead.

Reply:
I saw that, let me see if we can use it. If I recall when we needed `Offset` the projects were not yet integrated so I made a very similar local class.